File: fsharp/hello/HelloTest.fs
// M23 seed F# test; consumer of fsharp_test (plain executable: exit
// code is the verdict; the xUnit/NUnit runner selection stays open under
// O31).
module HelloTest

[<EntryPoint>]
let main _ =
    let got = Hello.greet "world"
    if got <> "hello world" then
        printfn "FAIL: got '%s'" got
        1
    else
        printfn "PASS"
        0

File: csharp/hello/HelloTest.cs
// M23 seed C# test; consumer of csharp_test (plain executable: exit
// code is the verdict; the xUnit/NUnit runner selection stays open under
// O31).
using System;

public static class HelloTestMain
{
    public static int Main(string[] args)
    {
        var got = Hello.Greeter.Greet("world");
        if (got != "hello world")
        {
            Console.WriteLine("FAIL: got '" + got + "'");
            return 1;
        }
        Console.WriteLine("PASS");
        return 0;
    }
}
